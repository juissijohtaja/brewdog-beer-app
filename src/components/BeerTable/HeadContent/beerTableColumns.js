const beerTableColumns = [
  {
    id: 'image',
    label: '',
    minWidth: 40,
  },
  {
    id: 'name',
    label: 'Beer',
    minWidth: 240
  },
  {
    id: 'abv',
    label: 'ABV',
    minWidth: 80,
    align: 'right',
  },
  {
    id: 'description',
    label: 'Description',
    minWidth: 180
  },
  {
    id: 'firstBrewed',
    label: 'First Brewed',
    minWidth: 160,
    align: 'right',
    sortable: true
  },
  {
    id: 'tagline',
    label: 'Tagline',
    minWidth: 140,
  },
  {
    id: 'brewersTips',
    label: 'Brewers Tips',
    minWidth: 180,
  },
  {
    id: 'ph',
    label: 'pH',
    minWidth: 60,
    align: 'right',
  },
  {
    id: 'foodPairing',
    label: 'Food Pairing',
    minWidth: 180,
  },
]

export default beerTableColumns