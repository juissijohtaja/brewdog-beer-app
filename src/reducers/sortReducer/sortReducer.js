const sortReducer = (state = true, action) => {
  switch (action.type) {
  case 'SET_SORT':
    return !state
  case 'RESET_SORT':
    return true
  default:
    return state
  }
}

export const sortChange = () => ({ type: 'SET_SORT' })
export const sortReset = () => ({ type: 'RESET_SORT' })

export default sortReducer