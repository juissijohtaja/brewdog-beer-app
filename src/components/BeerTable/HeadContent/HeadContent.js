import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import TableSortLabel from '@material-ui/core/TableSortLabel'
import TableCell from '@material-ui/core/TableCell'
import beerTableColumns from './beerTableColumns'
import { sortChange } from '../../../reducers/sortReducer/sortReducer'

const HeadContent = () => {
  const sortDesc = useSelector(state => state.sortDesc)
  const dispatch = useDispatch()
  const handleSort = () => dispatch(sortChange())

  return (
    <TableHead data-testid='head-content'>
      <TableRow>
        {beerTableColumns.map((column) => (
          <TableCell
            key={column.id}
            align={column.align}
            style={{ minWidth: column.minWidth }}
          >
            {column.sortable
              ? <TableSortLabel
                className='sortLabel'
                direction={sortDesc ? 'desc' : 'asc'}
                active
                onClick={handleSort}
              >
                {column.label}
              </TableSortLabel>
              : column.label
            }
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  )
}

export default HeadContent