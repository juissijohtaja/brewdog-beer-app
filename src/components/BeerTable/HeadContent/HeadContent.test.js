import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import initialStoreState from '../../../setupTests'
import HeadContent from './HeadContent'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { sortChange } from '../../../reducers/sortReducer/sortReducer'

describe('HeadContent snapshots and actions', () => {
  let store, component
  const mockStore = configureStore([])

  beforeEach(() => {
    store = mockStore(initialStoreState)
    store.dispatch = jest.fn()

    component = renderer.create(
      <Provider store={store}>
        <HeadContent />
      </Provider>
    )
  })

  test('should render correctly', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should dispatch 1 action on icon click', () => {
    renderer.act(() => {
      component.root.findByProps({ className: 'sortLabel' }).props.onClick()
    })

    expect(store.dispatch).toHaveBeenCalledTimes(1)
    expect(store.dispatch).toHaveBeenCalledWith(
      sortChange()
    )
  })
})