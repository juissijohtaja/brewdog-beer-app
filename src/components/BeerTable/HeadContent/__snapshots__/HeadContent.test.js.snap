// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadContent snapshots and actions should render correctly 1`] = `
<thead
  className="MuiTableHead-root"
  data-testid="head-content"
  role={null}
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
      scope="col"
      style={
        Object {
          "minWidth": 80,
        }
      }
    >
      Beer
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head"
      scope="col"
      style={
        Object {
          "minWidth": 180,
        }
      }
    >
      Description
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
      style={
        Object {
          "minWidth": 130,
        }
      }
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root sortLabel MuiTableSortLabel-active"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Start Date
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head"
      scope="col"
      style={
        Object {
          "minWidth": 140,
        }
      }
    >
      Category
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head"
      scope="col"
      style={
        Object {
          "minWidth": 150,
        }
      }
    >
      Responsible
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
      style={
        Object {
          "minWidth": 140,
        }
      }
    >
      Savings Amount
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head"
      scope="col"
      style={
        Object {
          "minWidth": 92,
        }
      }
    >
      Currency
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head"
      scope="col"
      style={
        Object {
          "minWidth": 130,
        }
      }
    >
      Complexity
    </th>
  </tr>
</thead>
`;
